darts/models/forecasting/nbeats.py

```diff
@@ -346,18 +407,18 @@ def forward(self, x):
             # set current stack residual as input for next stack
             x = stack_residual
 
-        # in multivariate case, we get a result x1, y1, z1 we want to reshape to original format
-        y = y.reshape(y.shape[0], self.output_chunk_length, self.input_dim)
-
-        # if some covariates, we don't want them for the output to be predicted.
-        # the covariates are by construction added as extra time series on the right side. So we need to get rid of this
-        # right output
-        y = y[:, :, :self.output_dim]
+        # In multivariate case, we get a result x1, y1, z1, x2, y2, z2, ... we want to reshape to original format
+        # In addition, we reshape y in order to have consecutive "blocks" containing the different parameters along the
+        # last dimension. We also get rid of the covariates and keep only the target dimensions.
+        # The covariates are by construction added as extra time series on the right side. So we need to get rid of this
+        # right output (keeping only :self.output_dim).
+        y = torch.cat([y[:,:,i].reshape(y.shape[0], self.output_chunk_length, self.input_dim)[:, :, :self.output_dim]
+                       for i in range(self.nr_likelihood_params)], dim=2)
```

Comment:
Mmmh I think you're right, but we may have an inconsistency here:
* `QuantileRegression` requires inputs (i.e. model outputs) to be of shape `(batch_size, n_timesteps, n_components, n_quantiles)` (it does the reshaping into this shape when sampling and computing the loss),
* The other `Likelihood` objects having several parameters assume a shape of `(batch_size, n_timesteps, n_components * n_params)` and they take consecutive blocks of the last dimensions as the parameters. For instance in the Gaussian case:
```
mu = model_output[:, :, :output_size // 2]
sigma = self.softplus(model_output[:, :, output_size // 2:])
```
I'm not sure that by just reshaping we get the right "blocks" in the right place, we might have to change the way either Likelihoods or QuantileRegressions take their parameters from the model outputs... (but I still have to think about it :D). Another solution (maybe quite clean) would be to force all models outputs to be of shape `(batch_size, n_timesteps, n_components, n_params)` rather than "abusing" the last dimension.

Reply:
I agree!
I vote for `(batch_size, n_timesteps, n_components, n_params)` as it's quite clean and similar to the shape we have for TimeSeries with samples. 
For this, we would have to unsqueeze `y_true` in `dim=-1` to compute the losses correctly for `MSELoss()` etc. as well (I believe currently `y_true` comes as `(batch_size, n_timesteps, n_components)` right now)

Reply:
OK, I'll give it a try later.